packages/node_modules/@ciscospark/internal-plugin-ediscovery/src/ediscovery.js

```diff
@@ -0,0 +1,520 @@
+/* eslint-disable prefer-template */
+/*!
+ * Copyright (c) 2015-2017 Cisco Systems, Inc. See LICENSE file.
+ */
+
+import {waitForValue, SparkPlugin} from '@ciscospark/spark-core';
+import reportRequest from './report-request';
+import ReportGenerator from './report-generator';
+
+/**
+ * @class Ediscovery is used by compliance officers to run compliance reports
+ *
+ */
+const Ediscovery = SparkPlugin.extend({
+  namespace: 'Ediscovery',
+
+  /**
+   * Creates a compliance report with a specific set of search parameters
+   *
+   * @param {String} name - A label to identify the report
+   * @param {String} description - A textual summary of the reports purpose
+   * @param {Array<String>} emails - A list of user emails relevant to the report
+   * @param {Array<String>} keywords - A list of search terms relevant to the report
+   * @param {Array<String>} spaceNames - A list of space names relevant to the report
+   * @param {Array<String>} spaceIds - A list of space ids relevant to the report
+   * @param {Object} range - Contains the start time and end time defining the search period
+   * @returns {Promise<ResponseEntity>} Http response containing the new report record
+   */
+  @waitForValue('@')
+  createReport(name, description, emails, keywords, spaceNames, spaceIds, range) {
```

Comment:
Once you get beyond a few arguments, it's typically better to just have the method accept an object. Helps with ordering and future proofing.

Reply:
Good idea, in fact, we already have a reportRequest object we could use: 
https://github.com/webex/spark-js-sdk/pull/1090/files#diff-bfd2923745106b14ceb4ea04fd757d5c